mkcodes.py

```diff
@@ -79,43 +113,55 @@ def get_files(inputs):
         elif path.suffix in markdown_extensions:
             yield path, path.parent
 
-def add_inits_to_dir(path):
+
+def add_inits_along_path(from_path, to_path):
     """Recursively add __init__.py files to a directory
-    This compensates for https://bugs.python.org/issue23882 and https://bugs.python.org/issue35617
+    This compensates for https://bugs.python.org/issue23882
+    and https://bugs.python.org/issue35617
     """
-    for child in path.rglob('*'):
-        if child.is_dir():
-            (child / '__init__.py').touch()
-
+    to_path = to_path.expanduser().resolve()
+    from_path = from_path.expanduser().resolve()
+    (to_path / '__init__.py').touch()
+    if to_path == from_path:
+        return
+    if to_path.relative_to(from_path):
+        if from_path != to_path.parent:
+            add_inits_along_path(from_path, to_path.parent)
 
 
 @click.command()
 @click.argument(
     'inputs', nargs=-1, required=True, type=click.Path(exists=True))
-@click.option('--output', default='{name}.py')
+@click.option('--output', default='{name}.{ext}')
 @click.option('--github/--markdown', default=bool(not markdown_enabled),
               help='Github-flavored fence blocks or pure markdown.')
 @click.option('--safe/--unsafe', default=True,
               help='Allow code blocks without language hints.')
 @click.option('--package-python', default=True,
-              help='Add __init__.py files to python output to aid in test discovery')
-def main(inputs, output, github, safe, package_python):
+              help='Add __init__.py files to python dirs for test discovery')
+@click.option('--default_lang', default='py',
+              help='Assumed language for code blocks without language hits.')
```

Comment:
There is a bit of dissonance between the `--safe` flag and the `--default-lang` option that we may want to resolve. Some ideas:

- Perhaps the use of both of these flags should raise an error? (I'd imagine there's an option to indicate incompatible arguments in the `click` api.)
- Perhaps we should remove the `--safe` flag in  favor of `--default_lang=` (no value). Maybe `--safe` should just be an alias for `--default_lang=`.

Reply:
sure - I lean towards --safe == default_lang==None
 
the motivation for safe seems to be "ignore code blocks that don't have a language!
default_lang's motivation is "assume the language is this if you don't know".
If there is no language and you don't know what extension to use, I can't imagine a scenario where you want to output the codes.  
If you have some directories where the language on unmarked codes is py and some directories where the default lang  is elm, you can indicate that easily with multiple runs of the same directory.

If you have a single file with multiple code blocks where you don't know the language, you really just need to get the document fixed to indicate the languages...